### PR TITLE
Cvsl 2051 cvl ca caseload filtering cases eligible for licence

### PR DESCRIPTION
**Pull request by nishanthmahasamudram**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/EligibilityService.kt [hunk 1]
```diff
@@ -25,6 +25,7 @@ class EligibilityService(
     hasActivePrisonStatus() describedAs "is not active in prison",
     hasReleaseDateInTheFuture() describedAs "release date in past",
     !isRecallCase() describedAs "is a recall case",
+    !isBreachOfTopUpSupervision() describedAs "is breach of top up supervision case",
```
Comment (nishanthmahasamudram) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/EligibilityService.kt:28]: added a method to filter out cases with imprisonmentStatus == "BOTUS".

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/CaCaseloadService.kt [hunk 2]
```diff
@@ -273,13 +269,7 @@ class CaCaseloadService(
     // Default status (if not overridden below) will show the case as clickable on case lists
     var licenceStatus = NOT_STARTED
 
-    if (isBreachOfTopUpSupervision(c)) {
```
Comment (nishanthmahasamudram) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/CaCaseloadService.kt:276]: removing these conditions as we are filtering out **recall and breachOfTopUpSupervision cases** in the eligibility check which is called before this.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/util/LicenceEventType.kt [hunk 3]
```diff
@@ -22,8 +22,6 @@ enum class LicenceEventType {
   VERSION_CREATED,
   NOT_STARTED,
   TIMED_OUT,
-  OOS_BOTUS,
```
Comment (nishanthmahasamudram) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/util/LicenceEventType.kt:25]: **note:** Not used anywhere.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/EligibilityServiceTest.kt [hunk 4]
```diff
@@ -216,6 +216,26 @@ class EligibilityServiceTest {
     assertThat(result).containsExactly("has no conditional release date")
   }
 
+  @Test
```
Comment (nishanthmahasamudram) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/EligibilityServiceTest.kt:219]: **note:** tests to prove cases with imprisonmentStatus == "BOTUS" is not eligible for CVL licence.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/EligibilityService.kt [hunk 5]
```diff
@@ -99,4 +100,8 @@ class EligibilityService(
     // Trust the Nomis recall flag as a fallback position - the above rules should always override
     return@early it.recall ?: error("${it.prisonerNumber} missing recall flag")
   }
+
+  private fun isBreachOfTopUpSupervision(): EligibilityCheck = early@{
+    it.imprisonmentStatus != null && it.imprisonmentStatus == "BOTUS"
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/EligibilityService.kt:105]: don't need the null check as `it.imprisonmentStatus == "BOTUS"` will equal false when `imprisonmentStatus` is null

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**